crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -312,6 +312,44 @@ def create_proposal
     base
   end
 
+  # Simple check if HA clusters report some problems
+  # If there are no problems, empty hash is returned.
+  # If this fails, information about failed actions for each cluster founder is
+  # returned in a hash that looks like this:
+  # {
+  #     "crm_failures" => {
+  #             "node1" => "reason for crm status failure"
+  #     },
+  #     "failed_actions" => {
+  #             "node2" => "Failed action on this node"
+  #     }
+  # }
+  # User has to manually clean pacemaker resources before proceeding with the upgrade.
+  def check_cluster_health
+    ret = {}
+    crm_failures = {}
+    failed_actions = {}
+
+    cluster_founders = NodeObject.find("pacemaker_founder:true AND pacemaker_config_environment:*")
+    return ret if cluster_founders.empty?
+    check_if_nodes_are_available cluster_founders
```

Comment:
Yes, it might have sense to move everything upgrade related out of this, not just this method